starlette/middleware/servertiming.py

```diff
@@ -0,0 +1,106 @@
+import threading
+import time
+import typing
+
+from starlette.middleware.base import BaseHTTPMiddleware, RequestResponseEndpoint
+from starlette.requests import Request
+from starlette.types import ASGIApp
+
+_thread_local = threading.local()
+
+
+class Ticker:
+    def __init__(self, name="app", description="main app"):
+        self.name = name
+        self.description = description
+        self._start = self._end = None
+
+    def start(self):
+        self._start = time.time()
+        _add_ticker(self)
+
+    def end(self):
+        self._end = time.time()
+
+    @property
+    def duration(self):
+        if self._start:
+            return int(((self._end or time.time()) - self._start) * 1000)
+        else:
+            return 0
+
+
+def get_tickers() -> typing.List[Ticker]:
+    return _thread_local.__dict__.setdefault("tickers", [])
```

Comment:
It's idiomatic to subclass threading.local and use a ClassVar to set the default

Reply:
Thanks for the review, corrected.